### ISSUE TITLE
`math.isinf.__self__` is None

### ISSUE THREAD

**Issue by mattip** [pypy/pypy#4959]
On PyPy:
```
>>>> math.isinf.__self__
>>>>
```

On CPython:
```
>>> math.isinf.__self__
<module 'math' (built-in)>
```

I think this is the root cause of https://github.com/annotated-types/annotated-types/issues/71

**Comment by cfbolz** [comment]
no, it should return the module *object*. I can do it, I just remembered how `BuiltinFunctions` work :sweat_smile:

**Comment by cfbolz** [comment]
wrote a prototype here: `py3.10-builtin-function-__self__`

**Comment by mattip** [comment]
I think this is needed, even if it is not the documented behavior of CPython.

**Comment by devdanzin** [comment]
Is this the same case as `__import__.__module__` being 'builtins' in CPython but `None` on PyPy (it's the only entry in the `builtins` module for which this is true)?

**Comment by cfbolz** [comment]
I've merged my fixes, we should try to see whether the original issue goes away with tomorrow's nightly.

**Comment by mattip** [comment]
The module is properly assigned on latest py3.10 HEAD, thanks @cfbolz. Closing.

**Comment by cfbolz** [comment]
should we backport this to 3.9 as well?

**Comment by mattip** [comment]
nice to have, not a release blocker.

**Comment by mattip** [comment]
We will not be releasing another py3.9, closing since this is fixed on 3.10
